source/main.cpp

```diff
@@ -95,7 +95,7 @@ int main(int argc, char *argv[])
 	// Ensure that we log errors to the errors.txt file.
 	Logger::SetLogErrorCallback([&hasErrors](const string &errorMessage) {
 		static const string PARSING_PREFIX = "Parsing: ";
-		if(errorMessage.substr(PARSING_PREFIX.length()) != PARSING_PREFIX)
+		if(errorMessage.substr(min(PARSING_PREFIX.length(), errorMessage.length())) != PARSING_PREFIX)
```

Comment:
The thing was wrong to begin with. It should be:
```suggestion
		if(errorMessage.substr(0, PARSING_PREFIX.length()) != PARSING_PREFIX)
```
No exception is thrown if `pos + count > size()` so no additional bounds checking is necessary.

https://en.cppreference.com/w/cpp/string/basic_string/substr

Reply:
https://github.com/endless-sky/endless-sky/commit/9215f8f7029ecfa84ac8c9747cda021b92e1c28e